src/network/ip_address_pool.cpp

```diff
@@ -112,6 +112,24 @@ mp::IPAddress mp::IPAddressPool::obtain_ip_for(const std::string& name)
     return it->second;
 }
 
+mp::IPAddress mp::IPAddressPool::check_ip_for(const std::string& name)
+{
+    const auto it = ip_map.find(name);
+    if (it == ip_map.end())
+    {
+        return mp::IPAddress{""};
+    }
+    return it->second;
+}
+
+mp::IPAddress mp::IPAddressPool::first_free_ip()
+{
+    if (ips_in_use.empty())
+        return start_ip;
+
+    return *ips_in_use.crbegin() + 1;
+}
```

Comment:
Based on the use, I don't think this will do the right thing.

Because old VM's could have a hole in their IP Assignments, not necessarily contiguous depending on the order of how other VM's were removed.

Exposing this "first free ip" should not be exposed externally.

Reply:
What I'm trying to accomplish here is to figure out which IP for dnsmasq to begin handing out dynamic IP's.  Since ips_in_use is a std::set, then even if there are holes in assignment, the IP's should be in order, so if I take the very last one in the set and add 1, then I'm getting the first IP in the range that dnsmasq can use.

For example, say the set has 10.122.122.2, 10.122.122.4, and 10.122.122.6, then this should return 10.122.122.7 and I can set dnsmasq to use the range of 10.122.122.7 to 10.122.122.254.  I really don't care that there are holes, all I'm trying to get is a range.

Reply:
OK, based on IRC conversation it should be ok with the caveat that a big gap in IP addresses could exist if the user created many VM's then deleted VM's in between but we'll assume that would be a rare occurrence.